product_code/certificate_service/api/src/main/scala/de/upb/cs/uc4/certificate/model/EncryptedPrivateKey.scala

```diff
@@ -1,8 +1,40 @@
 package de.upb.cs.uc4.certificate.model
 
+import de.upb.cs.uc4.shared.client.exceptions.SimpleError
 import play.api.libs.json.{ Format, Json }
 
-case class EncryptedPrivateKey(key: String, iv: String, salt: String)
+import scala.concurrent.{ ExecutionContext, Future }
+
+case class EncryptedPrivateKey(key: String, iv: String, salt: String) {
+
+  def validate(implicit ec: ExecutionContext): Future[Seq[SimpleError]] = Future {
+    val keyRegex = """[\s\S]{1,8192}""".r
+    val ivRegex = """[\s\S]{1,64}""".r
+    val saltRegex = """[\s\S]{1,256}""".r
+
+    if (key.isEmpty && iv.isEmpty && salt.isEmpty) {
+      Seq()
+    }
+    else if (Seq(key, iv, salt).contains("")) {
+      Seq(SimpleError("encryptedPrivateKey", "Either all fields must be empty or no fields must be empty."))
+    }
+    else {
+      var errors = List[SimpleError]()
+
+      if (!keyRegex.matches(key)) {
+        errors :+= SimpleError("key", "The key must not be empty or longer than 8192 characters.")
```

Comment:
Since an empty key field in an of itself is not wrong, the message is misleading. I suggest another message
```suggestion
        errors :+= SimpleError("key", "For a non-empty key object, the key must not be longer than 8192 characters.")
```